executor/distsql_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor_test
+
+import (
+	"bytes"
+	"fmt"
+	"runtime/pprof"
+	"strings"
+	"time"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/executor"
+	"github.com/pingcap/tidb/store/tikv"
+	"github.com/pingcap/tidb/util/testkit"
+)
+
+// This test checks that when a index double read returns before reading all the rows, the goroutine doesn't
+// leak. For testing distsql with multiple regions, we need to manually split a mock TiKV.
+func (s *testSuite) TestIndexDoubleReadClose(c *C) {
+	if _, ok := s.store.GetClient().(*tikv.CopClient); !ok {
+		// Make sure the store is tikv store.
```

Comment:
The test can be configured to run with localstore, then the client is not `*tikv.CopClient`.